commit 71fa8ce7ed133efb19dacd99d71a15abc357a5ca (HEAD -> master, refs/changes/83/1883/3)
Author: David Pacheco <dap@joyent.com>
Date:   2017-05-02T06:59:57-07:00 (2 years, 5 months ago)
    
    MANTA-3242 gen-manta-net reports spurious missing uuid warning
    Reviewed by: Richard Bradley <richard.bradley@joyent.com>
    Approved by: Richard Bradley <richard.bradley@joyent.com>

diff --git a/bin/mgenazconfig.js b/bin/mgenazconfig.js
index 92a3dc5..b57f679 100755
--- a/bin/mgenazconfig.js
+++ b/bin/mgenazconfig.js
@@ -1063,13 +1063,6 @@ function mgGenMantaNet(mgopts, callback)
 		mgopts.mgo_devices_by_az[az.name].forEach(function (device) {
 			var type, uuid;
 
-			if (device.d42d_uuid === null) {
-				uuid = device.d42d_serial;
-				counters_az['nMissingUuid']++;
-			} else {
-				uuid = device.d42d_uuid;
-			}
-
 			if (racks.indexOf(device.d42d_rack) < 0)
 				return;
 
@@ -1078,6 +1071,13 @@ function mgGenMantaNet(mgopts, callback)
 			    device.d42d_hardware))
 				return;
 
+			if (device.d42d_uuid === null) {
+				uuid = device.d42d_serial;
+				counters_az['nMissingUuid']++;
+			} else {
+				uuid = device.d42d_uuid;
+			}
+
 			type = mgHardwareToServerType[device.d42d_hardware];
 			if (type !== 'storage' && type !== 'metadata')
 				return;
